src/File.php

```diff
@@ -397,6 +397,14 @@ public function canView($member = null)
             return $member->inGroups($this->ViewerGroups());
         }
 
+        // Specific users can view this file
+        if ($this->CanViewType === InheritedPermissions::ONLY_THESE_MEMBERS) {
+            if (!$member) {
+                return false;
+            }
+            return \in_array($member->ID, $this->ViewerMembers()->column('ID'), true);
```

Comment:
as per https://github.com/silverstripe/silverstripe-cms/pull/2862#discussion_r1253699612
```suggestion
            return $this->ViewerMembers()->filter('ID', $member->ID)->count() > 0;
```